Map/MapManager.gd

```diff
@@ -4,18 +4,14 @@ extends Node2D
 static func create_map(): ## Generates and Populates a map with rooms that have random room types. More in depth algorithms will be added in the future
 	var _map: MapBase = MapBase.new()
 	var _grid: Array = [] ## 2d array to return. this will be populated with rooms
-	
-	for i: int in _map.width.size(): ## construct 2d array to return
+	for index_width: int in range(_map.floors_width.size()): ## loop through elements of the grid and assign a room type
```

Comment:
this should be the opposite actually, `index_height`. Since the array gives the different width, the array is being read iterating through the heights.